### PR TITLE
coqPackages.reglang: init at 1.1.2

### PR DESCRIPTION
**Pull request by siraben**
###### Motivation for this change
Add https://github.com/coq-community/reglang

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 126106` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>coqPackages.reglang</li>
  </ul>
</details>

File: pkgs/development/coq-modules/reglang/default.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  releaseRev = v: "v${v}";
+
+  release."1.1.2".sha256 = "sha256-SEnMilLNxh6a3oiDNGLaBr8quQ/nO2T9Fwdf/1il2Yk=";
+
+  inherit version;
+  defaultVersion = with versions; switch coq.coq-version [
+    { case = isEq "8.13"; out = "1.1.2"; }
+    { case = isEq "8.12"; out = "1.1.2"; }
+    { case = isEq "8.11"; out = "1.1.2"; }
+    { case = isEq "8.10"; out = "1.1.2"; }
```
Comment (vbgl) [inline pkgs/development/coq-modules/reglang/default.nix:16]: Maybe you can use a range of versions here.

File: pkgs/development/coq-modules/reglang/default.nix [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  ] null;
+
+  # mlPlugin = true;
```
Comment (vbgl) [inline pkgs/development/coq-modules/reglang/default.nix:19]: Is this comment useful?

File: pkgs/development/coq-modules/reglang/default.nix [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+  propagatedBuildInputs = [ mathcomp ];
```
Comment (vbgl) [inline pkgs/development/coq-modules/reglang/default.nix:21]: Do you really need all of `mathcomp`? The `ssreflect` part seems enough.

File: pkgs/top-level/coq-packages.nix [hunk 4]
```diff
@@ -70,6 +70,7 @@ let
       pocklington = callPackage ../development/coq-modules/pocklington {};
       QuickChick = callPackage ../development/coq-modules/QuickChick {};
       relation-algebra = callPackage ../development/coq-modules/relation-algebra {};
+      reglang = callPackage ../development/coq-modules/reglang {};
```
Comment (vbgl) [inline pkgs/top-level/coq-packages.nix:73]: Please try to list the packages in alphabetical order…

Comment (siraben) [reply pkgs/top-level/coq-packages.nix:73]: Oops, my bad.

File: pkgs/development/coq-modules/reglang/default.nix [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+mkCoqDerivation {
+  pname = "reglang";
+
+  releaseRev = v: "v${v}";
+
+  release."1.1.2".sha256 = "sha256-SEnMilLNxh6a3oiDNGLaBr8quQ/nO2T9Fwdf/1il2Yk=";
+
+  inherit version;
+  defaultVersion = with versions; switch coq.coq-version [
+    { case = pred.inter (isGe "8.10") (isLe "8.13"); out = "1.1.2"; }
```
Comment (CohenCyril) [inline pkgs/development/coq-modules/reglang/default.nix:13]:
```suggestion
    { case = range "8.10" "8.13"; out = "1.1.2"; }
```

**Pull request merged**